### PR TITLE
177147970 upload when launch from portal

### PR DESCRIPTION
**Pull request by knowuh**
@dougmartin -- I just wanted to let you know that this PR is here, and getting large.  

I don't think it's ready to merge yet, but I think a glance might help you know where I am at.

There are changes here that might inform what to do WRT remote endpoints for the user for log uploading.

### CODE REVIEW

File: src/components/activity-completion/completion-export-answers.tsx [hunk 1]
```diff
@@ -16,7 +16,8 @@ export class CompletionExportAnswers extends React.PureComponent<IProps, IState>
     this.state = {activityJSON: "", filename: "activity"};
   }
   async componentDidMount() {
-    const activityJSONComplete = await Storage.exportActivityToJSON();
+    const storage = await getStorage();
```
Comment (dougmartin) [inline src/components/activity-completion/completion-export-answers.tsx:19]: I don't think you need the `await` here `getStorage` does not return a promise.

**Review by dougmartin — COMMENTED** [review]
Looks good so far :+1:

**Comment by dougmartin** [comment]
I rebased this to the `offline-mode` branch and there were a lot of conflicts.  I believe I have resolved them all - I had to force push to update the branch.

**Comment by dougmartin** [comment]
I decided to remove the rebase commit and reset to Scott's last commit.  I'm going to do a merge instead as there were a lot of merge errors and I think the merge commit should be reviewed (this is also based on updating a follow on PR based on this branch and seeing a lot of merge errors that also need to be reviewed)

**Comment by scytacki** [comment]
I reviewed the merge commit and it looks fine to me.

**Review by scytacki — APPROVED** [review]
looks good, just one issue that I'll look at later.

**Pull request merged**